service/telemetry.go

```diff
@@ -109,8 +110,32 @@ func (tel *colTelemetry) initOnce(col *Collector) error {
 
 	logger.Info("Setting up own telemetry...")
 
-	instanceUUID, _ := uuid.NewRandom()
-	instanceID := instanceUUID.String()
+	if resource == nil {
+		resource = map[string]*string{}
+	}
+
+	if _, ok := resource[semconv.AttributeServiceInstanceID]; !ok {
+		// AttributeServiceInstanceID is not specified in the config. Auto-generate one.
+		instanceUUID, _ := uuid.NewRandom()
+		instanceID := instanceUUID.String()
+		resource[semconv.AttributeServiceInstanceID] = &instanceID
+	}
+
+	if AddCollectorVersionTag {
+		if _, ok := resource[semconv.AttributeServiceVersion]; !ok {
+			// AttributeServiceVersion is not specified in the config. Use the actual
+			// build version.
+			resource[semconv.AttributeServiceVersion] = &version.Version
+		}
+	}
+
+	// Empty Resource attributes are an indication by the user to suppress and stop
+	// emitting the particular attribute.
+	for k, v := range resource {
+		if v == nil {
+			delete(resource, k)
+		}
+	}
```

Comment:
I think we should use new `map[string]string` instead of mutating original collector configuration object (including this place https://github.com/open-telemetry/opentelemetry-collector/pull/5402/files#diff-52da5566fa270d6a253f7a301b66f4a1dbda7c2c42f3781f8357efe7f08dff10R121). Then we can also remove this redundant check https://github.com/open-telemetry/opentelemetry-collector/pull/5402/files#diff-52da5566fa270d6a253f7a301b66f4a1dbda7c2c42f3781f8357efe7f08dff10R206.

WDYT?

Reply:
Done.